File: src/main/java/com/gmail/borlandlp/minigamesdtools/arena/ArenaCreatorHub.java
package com.gmail.borlandlp.minigamesdtools.arena;

import com.gmail.borlandlp.minigamesdtools.creator.AbstractDataProvider;
import com.gmail.borlandlp.minigamesdtools.creator.CreatorHub;

public class ArenaCreatorHub extends CreatorHub {
    public ArenaBase createArena(String ID, AbstractDataProvider dataProvider) throws Exception {
        return (ArenaBase) this.create(ID, dataProvider);
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/arena/ArenaCreatorHub.kt
package com.gmail.borlandlp.minigamesdtools.arena

import com.gmail.borlandlp.minigamesdtools.creator.AbstractDataProvider
import com.gmail.borlandlp.minigamesdtools.creator.CreatorHub

class ArenaCreatorHub : CreatorHub() {
    @Throws(Exception::class)
    override fun create(itemID: String, dataProvider: AbstractDataProvider): ArenaBase {
        return super.create(itemID, dataProvider) as ArenaBase
    }
}